### PR TITLE
Add axisymmetry capability to interaction integral for SIF 

### LINKED ISSUE #23631: Implement and verify interaction integral for axisymmetric problem
## Reason
Have the ability to compute stress intensity factors via interaction integral for problems that use `RZ` symmetry.

## Design
No significant design changes are expected.

## Impact
Added feature.

### PR DESCRIPTION
**Pull request by recuero**
Closes #23631

### CODE REVIEW

File: modules/tensor_mechanics/doc/content/modules/tensor_mechanics/FractureIntegrals.md [hunk 1]
```diff
@@ -74,7 +74,18 @@ The interaction integral method is based on the $J$-integral and makes it possib
...
+
+
+!alert note
```
Comment (bwspenc) [inline modules/tensor_mechanics/doc/content/modules/tensor_mechanics/FractureIntegrals.md:80]: In my opinion, this probably doesn't need to be an alert.

File: modules/tensor_mechanics/include/vectorpostprocessors/InteractionIntegral.h [hunk 2]
```diff
@@ -66,10 +66,13 @@ class InteractionIntegralTempl : public ElementVectorPostprocessor
    * @param aux_stress Auxiliary stress -- computed in this method
    * @param grad_disp Gradient of auxiliary displacement -- computed in this method
    * @param grad_disp Auxiliary strain (FGM) -- computed in this method
+   * @param aux_disp Auxiliary displacement (RZ) -- computed in this method
```
Comment (bwspenc) [inline modules/tensor_mechanics/include/vectorpostprocessors/InteractionIntegral.h:69]: Maybe explicitly say that the aux displacements are only nonzero for RZ. In general, they are actually nonzero all the time, but we just don't use them, right?

**Review by bwspenc — DISMISSED** [review]
Looks good. I made a couple of minor notes, but those are optional.

File: modules/tensor_mechanics/include/vectorpostprocessors/InteractionIntegral.h (hunk 2 above)
Comment (recuero) [reply modules/tensor_mechanics/include/vectorpostprocessors/InteractionIntegral.h:69]: Yes, different terms of the interaction integral make use of different fields.

**Review by bwspenc — APPROVED** [review]

**Pull request merged**